app/views/bookmarks/_tools.html.erb

```diff
@@ -0,0 +1,11 @@
+<%# local override to add CSV link %>
+<ul class="<%= controller_name %>Tools nav nav-pills">
+  <%= render_show_doc_actions document_list, document: nil, document_list: @document_list, url_opts: Blacklight::Parameters.sanitize(params.to_unsafe_h) do |config, inner| %>
```

Comment:
Can't you do this via config somehow? Something like `add_show_tools_partial(:csv, callback: :csv_action)`? Similar to Email maybe. See https://github.com/projectblacklight/blacklight/blob/master/app/controllers/concerns/blacklight/default_component_configuration.rb#L17 and https://github.com/projectblacklight/blacklight/blob/master/app/controllers/concerns/blacklight/catalog.rb#L243.

Normally I'd be all for not doing it the blacklight way, but we'll have to upgrade at some point.

Reply:
@tpendragon the action / callback thing makes the csv show up in a modal. How does this look, with the smaller partial?

Reply:
Great!